src/Platform/Microsoft.Testing.Platform/IPC/DotnetTestDataConsumer.cs

```diff
@@ -32,13 +36,144 @@ public DotnetTestDataConsumer(NamedPipeClient dotnetTestPipeClient)
 
     public string DisplayName => nameof(DotnetTestDataConsumer);
 
-    public string Description => "Send back to the dotnet test informations";
+    public string Description => "Send back information to the dotnet test";
 
-    public Task ConsumeAsync(IDataProducer dataProducer, IData value, CancellationToken cancellationToken) => Task.CompletedTask;
+    public async Task ConsumeAsync(IDataProducer dataProducer, IData value, CancellationToken cancellationToken)
+    {
+        if (value is TestNodeUpdateMessage testNodeUpdateMessage)
```

Comment:
Use switch instead of if/elseif as you check only types, that's easier for cognitive load.